component/query.libsonnet

```diff
@@ -44,9 +47,101 @@ local query = thanos.query(queryBaseConfig + params.commonConfig + params.query
       type: params.query.serviceType,
     },
   },
+  serviceAccount+: {
+    metadata+: {
+      annotations+: if params.queryRbacProxy.enabled then {
+        'serviceaccounts.openshift.io/oauth-redirecturi.primary': params.queryRbacProxy.redirectUri,
+        // there is also another annotation:
+        //  serviceaccounts.openshift.io/oauth-redirectreference.primary: '{"kind":"OAuthRedirectReference","apiVersion":"v1","reference":{"kind":"Route","name":"<route-name>"}}'
+        // However, using an Ingress object generates a randomized Route name, and setting `kind: Ingress` didn't work.
+        // So we have to give an static URL.
+      },
+    },
+  },
+  deployment+: {
+    spec+: {
+      template+: {
+        spec+: {
+          containers+: if params.queryRbacProxy.enabled then [
+            {
+              name: 'openshift-oauth-proxy',
+              image: '%s/%s:%s' % [ proxyImage.registry, proxyImage.image, proxyImage.tag ],
+              args: [
+                '--upstream=http://0.0.0.0:9090',
+                '--provider=openshift',
+                '--cookie-secret-file=/var/run/secrets/kubernetes.io/serviceaccount/token',
+                '--openshift-service-account=%s' % 'thanos-query',
```

Comment:
The following should work (IIRC how `super` works):

```suggestion
                '--openshift-service-account=%s' % super.spec.template.spec.serviceAccountName,
```

Otherwise you may have to do the following before the `spec+` on line 62:

```
local self = super;
```

and then use `self.spec.template.spec.serviceAccountName`.

Reply:
Your suggestion doesn't work:
```
Jsonnet error: failed to compile /tmp/component-m237v6r3/dependencies/thanos/component/main.jsonnet:
 RUNTIME ERROR: attempt to use super when there is no super class.
	/tmp/component-m237v6r3/dependencies/thanos/component/query.libsonnet:73:52-57	thunk <b>
```
And `local self = super;` `make format_jsonnet` yields
```
STATIC ERROR: component/query.libsonnet:62:11-15: expected token IDENTIFIER but got (self, "self")
```
```
  deployment+: {
    local self = super;
    spec+: {
```

Reply:
So, turns out I was mixing up two features, what works is the following:

In `deployment.spec.template.spec`:

```
local pod = self;
{
  containers+: [ ... ]
}
```

and then use `pod.spec.serviceAccountName` when configuring the sidecar.

Reply:
that worked, thanks!